portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,47 @@
 
 package com.google.sps.servlets;
 
+import com.google.sps.data.CommentData;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
+import java.util.ArrayList;
+import com.google.gson.Gson;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    private CommentData commentlog = new CommentData();
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {  
+      // Convert the comments to JSON
+      String json = convertToJsonUsingGson(commentlog);
+
+      // Send the JSON as the response
+      response.setContentType("text/html");
```

Comment:
I think this should be application/json instead of test/html since you're sending json back now?